### PR TITLE
[FEAT] Contact Us API Endpoint

### PR DESCRIPTION
**Pull request by amowogbaje**
## Description

This PR introduces an API endpoint to handle user inquiries from the "Contact Us" form. The endpoint validates input data and sends the inquiry to a specified email address using the configured email service.

## Related Issue (Link to Github issue)

[Issue #4](https://github.com/hngprojects/hng_boilerplate_php_laravel_web/issues/4)

## Motivation and Context

This change is required to provide a backend service to handle user inquiries from the "Contact Us" form, ensuring data is validated and inquiries are sent to the appropriate email address.

#### Prerequisites:
- Ensure you have XAMPP installed on your system.
- Locate your `htdocs` or `www` folder in your XAMPP/WAMP installation directory.

#### Cloning the Repository
1. Clone the repository:
   ```sh
   git clone https://github.com/Goldengide/hng_boilerplate_php_laravel 
   ```
2. Change into the project directory:
   ```sh
   cd hng_boilerplate_php_laravel
   ```
3. Checkout the `main` branch:
   ```sh
   git checkout main
   ```

#### Running the Project
1. Install the dependencies:
   ```sh
   composer install
   ```
2. Generate the application key:
   ```sh
   php artisan key:generate
   ```
3. Copy the example environment file:
   ```sh
   cp .env.example .env
   ```
4. Update the following variables in the `.env` file with your email service credentials:
   ```
   MAIL_MAILER=smtp
   MAIL_HOST=smtp.gmail.com
   MAIL_PORT=587
   MAIL_USERNAME=your_username@gmail.com
   MAIL_PASSWORD=your_password
   MAIL_ENCRYPTION=tls
   MAIL_FROM_ADDRESS=your_username@gmail.com
   MAIL_FROM_NAME="${APP_NAME}"
   ```

### Unit Testing
1. Run the unit tests:
   ```sh
   php artisan test
   ```

### Testing the API
1. Use POSTMAN to send a POST request to `/api/v1/contact` with the following JSON body:
   ```json
   {
       "name": "Gideon Amowogbaje",
       "email": "gideon@james.com",
       "message": "This is a test message."
   }
   ```

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

---

### CODE REVIEW

**Comment by Ruthiejayjay** [comment]
Fix merge conflicts

**Comment by amowogbaje** [comment]
> Fix merge conflicts

I have fixed merge conflicts @Ruthiejayjay

File: app/Http/Controllers/ContactController.php
Comment (Ruthiejayjay) [inline app/Http/Controllers/ContactController.php:1]: use proper naming convention. status should be status_code

**Comment by Ruthiejayjay** [comment]
[Goldengide](https://github.com/Goldengide) fix the reviews

**Comment by amowogbaje** [comment]
Ok fixed

**Review by Ruthiejayjay — APPROVED** [review]

**Review by amowogbaje — COMMENTED** [review]
done

**Comment by Ruthiejayjay** [comment]
You're almost there, but all checks are failing. fix that

**Comment by amowogbaje** [comment]
finally @Ruthiejayjay

**Pull request merged**